aws/src/main/java/org/apache/iceberg/aws/s3/VendedCredentialsProvider.java

```diff
@@ -84,6 +85,7 @@ private RESTClient httpClient() {
 
   private LoadCredentialsResponse fetchCredentials() {
     return httpClient()
+        .withAuthSession(AuthSession.EMPTY)
```

Comment:
I just took a quick look because I'm also working on the HttpClient code and was curious about this PR.

This might have been commented, but wouldn't it be possible to pass the AuthSession to the HttpClient during creation, maybe through a Builder? Then we could avoid adding these 'withAuthSession' calls when we use the httpClient.

Reply:
Different sessions are used depending on the context, so it can't be restricted to a single one.